### PR TITLE
K8SPG-307: Merge upstream 5.3.1 changes

### PR DESCRIPTION
**Pull request by egegunes**
Merge changes from upstream 5.3.1 release

Upstream HEAD: f060545a3d7dd829a1b7ece82498e177ebbba228

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-379/b74051498/init-deploy.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-379/b74051498/demand-backup.log) |
| start-from-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-379/b74051498/start-from-backup.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-379/b74051498/monitoring.log) |
| We run 4 out of 4|

commit: https://github.com/percona/percona-postgresql-operator/pull/379/commits/b74051498749b847d385f923813832c30a73a81e
image: `perconalab/percona-postgresql-operator:PR-379-b74051498`

**Review by inelpandzic — APPROVED** [review]

File: .github/actions/k3d/action.yaml [hunk 1]
```diff
@@ -0,0 +1,94 @@
+name: k3d
```
Comment (hors) [inline .github/actions/k3d/action.yaml:1]: @egegunes should we have all these actions?

File: .github/workflows/codeql-analysis.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+name: CodeQL
+
```
Comment (hors) [inline .github/workflows/codeql-analysis.yaml:2]: the same for workflow

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -54,4 +54,4 @@ See the [Documentation Contribution Guide](https://github.com/percona/k8spg-docs
 
 ## Code review
 
-Your contribution will be reviewed by other developers contributing to the project. The more complex your changes are, the more experts will be involved. You will receive feedback and recommendations directly on your pull request on GitHub, so keep an eye on your submission and be prepared to make further amendments. The developers might even provide some concrete suggestions on modifying your code to match the project’s expectations better.
```
Comment (hors) [inline CONTRIBUTING.md:57]: @fiowro  we need to have own CONTRIBUTING guide

**Review by hors — CHANGES_REQUESTED** [review]

File: .github/actions/k3d/action.yaml (hunk 1 above)
Comment (egegunes) [reply .github/actions/k3d/action.yaml]: https://github.com/percona/percona-postgresql-operator/pull/379/commits/bc785d590629a5f3d85ee15b6dd5a07deb003165

File: .github/workflows/codeql-analysis.yaml (hunk 2 above)
Comment (egegunes) [reply .github/workflows/codeql-analysis.yaml]: https://github.com/percona/percona-postgresql-operator/pull/379/commits/bc785d590629a5f3d85ee15b6dd5a07deb003165

**Pull request merged**